### PR TITLE
Fixes #24102 - Allow more complex logic for template locks

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: app/models/template.rb [hunk 1]
```diff
@@ -114,6 +114,8 @@ def self.parse_metadata(text)
   #   :force - set to true if you want to bypass locked templates
   #   :associate - either 'new', 'always' or 'never', determines when the template should associate objects based on metadata
   #   :lock - lock imported templates (false by default)
+  #   :lock_predicate - a predicate that evaluates to boolean, determines whether a template should be locked
+  #                   - use instead of :lock when more complex logic for locking is required
```
Comment (stbenjam) [inline app/models/template.rb]: Wouldn't it be more railsy to just use `lock`?


e.g. 
```ruby
options[:lock].respond_to?(:call) ? options[:lock].call(self) : !!options[:lock]
```

So you could pass:

```ruby
{lock: true}
```
or

```ruby
{lock: {|t| !t.new_record? && t.locked }
```

**Review by dLobatog — DISMISSED** [review]
Setting on WoC based on @stbenjam 's comment

**Comment by xprazak2** [comment]
I made an update as suggested + minor refactoring to appease code climate.

**Review by stbenjam — DISMISSED** [review]
LGTM! Thanks

**Comment by stbenjam** [comment]
Bah, tests failed. There's a slight behavior change in this PR, when lock was nil before we previously didn't change the lock on an existing template, but now it'd be updated to be false.

**Comment by xprazak2** [comment]
Updated, tests should pass now.

File: app/models/template.rb [hunk 2]
```diff
@@ -179,6 +179,10 @@ def import_locations(options)
     end
   end
 
+  def handle_lock_on_import(options)
+    (self.locked = options[:lock].respond_to?(:call) ? options[:lock].call(self) : options[:lock]) unless options[:lock].nil?
```
Comment (dLobatog) [inline app/models/template.rb:211]: Could you add a couple of tests that test locking using a proc or a boolean?

**Review by dLobatog — DISMISSED** [review]
Left a comment inline, thanks @xprazak2

**Comment by xprazak2** [comment]
I added tests to make sure template is locked regardless of whether boolean or lambda is used.

**Comment by xprazak2** [comment]
[test katello]

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, díky @xprazak2!

**Pull request merged**